Psychtoolbox/PsychHardware/EyelinkToolbox/EyelinkDemos/SR-ResearchDemos/StereoPicture/EyeLink_StereoPicture.m

```diff
@@ -0,0 +1,408 @@
+        [~, vnumcell] = regexp(versionstring,'.*?(\d)\.\d*?','Match','Tokens'); % Extract EL version before decimal point
+        ELsoftwareVersion = str2double(vnumcell{1}{1}); % Returns 1 for EyeLink I, 2 for EyeLink II, 3/4 for EyeLink 1K, 5 for EyeLink 1KPlus, 6 for Portable Duo         
+        % Print some text in Matlab's Command Window
+        fprintf('Running experiment on %s version %d\n', versionstring, ver );
+    end
+    % Add a line of text in the EDF file to identify the current experimemt name and session. This is optional.
+    % If your text starts with "RECORDED BY " it will be available in DataViewer's Inspector window by clicking
+    % the EDF session node in the top panel and looking for the "Recorded By:" field in the bottom panel of the Inspector.
+    preambleText = sprintf('RECORDED BY Psychtoolbox demo %s session name: %s', mfilename, edfFile);
+    Eyelink('Command', 'add_file_preamble_text "%s"', preambleText);
+    
+    
+    %% STEP 2: SELECT AVAILABLE SAMPLE/EVENT DATA
+    % See EyeLinkProgrammers Guide manual > Useful EyeLink Commands > File Data Control & Link Data Control
+    
+    % Select which events are saved in the EDF file. Include everything just in case
+    Eyelink('Command', 'file_event_filter = LEFT,RIGHT,FIXATION,SACCADE,BLINK,MESSAGE,BUTTON,INPUT');
+    % Select which events are available online for gaze-contingent experiments. Include everything just in case
+    Eyelink('Command', 'link_event_filter = LEFT,RIGHT,FIXATION,SACCADE,BLINK,BUTTON,FIXUPDATE,INPUT');
+    % Select which sample data is saved in EDF file or available online. Include everything just in case
+    if ELsoftwareVersion > 3  % Check tracker version and include 'HTARGET' to save head target sticker data for supported eye trackers
+        Eyelink('Command', 'file_sample_data  = LEFT,RIGHT,GAZE,HREF,RAW,AREA,HTARGET,GAZERES,BUTTON,STATUS,INPUT');
+        Eyelink('Command', 'link_sample_data  = LEFT,RIGHT,GAZE,GAZERES,AREA,HTARGET,STATUS,INPUT');
+    else
+        Eyelink('Command', 'file_sample_data  = LEFT,RIGHT,GAZE,HREF,RAW,AREA,GAZERES,BUTTON,STATUS,INPUT');
+        Eyelink('Command', 'link_sample_data  = LEFT,RIGHT,GAZE,GAZERES,AREA,STATUS,INPUT');
+    end
+    
+    
+    %% STEP 3: OPEN GRAPHICS WINDOW IN STEREO MODE
+    
+    % Open experiment graphics
+    if isempty(screenNumber)
+        screenNumber = max(Screen('Screens'));
+    end
+    
+    if stereoMode == 10
+        % Yes. Do we have at least two separate displays for both views?
+        if length(Screen('Screens')) < 2
+            error('Sorry, for stereoMode 10 you''ll need at least 2 separate display screens in non-mirrored mode.');
+        end        
+        if ~IsWin % Assign left-eye view (the master window) to main display:
+            screenNumber = 0;
+        else
+            screenNumber = 1;
+        end
+    end
+    [window, ~] = Screen('OpenWindow', screenNumber, [128 128 128], [], [], [], stereoMode);

```

Comment:
This is **really** where you want to copy the approach shown in `ImagingStereoDemo.m` and the setup code therein, to get the maximum number of supported stereo modes at the highest quality with the best performance!

This here is not how it should be done anymore in any new demo.

Reply:
understood and agreed the StereoMode demo needs to be revamped to reflect the most current PTB conventions. I will include this as a priority item in the next round of revisions following completion of the rest of this pull request to allow for additional time to make those changes and thorough testing before publishing